src/main/java/org/mastodon/mamut/clustering/util/Classification.java

```diff
@@ -84,6 +92,9 @@ public Classification( final List< Pair< Set< T >, Cluster > > classifiedObjects
 		}
 		this.rootCluster = rootCluster;
 		this.cutoff = cutoff;
+
+		double[] nonZeroValues = Stream.of( distances ).flatMapToDouble( DoubleStream::of ).filter( value -> value != 0 ).toArray();
```

Comment:
Filtering all the zero values will remove too many values. Imaging having a forest of 99 identical trees and one largely different tree. The median should for this example be zero, but you will get a higher value...

Using the upper triangle of the matrix, or all non diagonal values would give the proper result.

Reply:
Thanks for this advice. I introduced a new method that explicitly extracts the upper triangle from the original matrix.